cpp/src/arrow/array/array_base.h

```diff
@@ -86,16 +86,17 @@ class ARROW_EXPORT Array {
   std::shared_ptr<DataType> type() const { return data_->type; }
   Type::type type_id() const { return data_->type->id(); }
 
-  /// Buffer for the null bitmap.
+  /// Buffer for the validity (null) bitmap, if any. Note that Union types
+  /// never have a null bitmap.
   ///
-  /// Note that for `null_count == 0`, this can be null.
-  /// This buffer does not account for any slice offset
+  /// Note that for `null_count == 0` or for null or union types, this will be
```

Comment:
Well, `null_count` would be 0 for a union array anyway, no?

Reply:
Right, will clarify